src/gradient/gradient.class.ts

```diff
@@ -1,493 +1,342 @@
-//@ts-nocheck
 
-import { Color } from "../color";
 
+import { fabric } from "../../HEADER";
+import { Color } from "../color";
+import { iMatrix } from "../constants";
+import { parseTransformAttribute } from "../parser/parseTransformAttribute";
+import { matrixToSVG, populateWithProperties } from "../util";
+import { parseColorStops, parseCoords, parseGradientUnits, parseType } from "./parser";
+import { ColorStop, GradientCoords, GradientOptions, GradientType, GradientUnits, SVGOptions } from "./typedefs";
+
+/**
+ * @todo remove this transient junk
+ */
+type FabricObject = any;
+
+/**
+ * Gradient class
+ * @class Gradient
+ * @tutorial {@link http://fabricjs.com/fabric-intro-part-2#gradients}
+ */
+export class Gradient<S, T extends GradientType = S extends GradientType ? S : 'linear'> {
 
-(function(global) {
-  var fabric = global.fabric;
-  /* _FROM_SVG_START_ */
-  function getColorStop(el, multiplier) {
-    var style = el.getAttribute('style'),
-        offset = el.getAttribute('offset') || 0,
-        color, colorAlpha, opacity, i;
+  /**
+   * Horizontal offset for aligning gradients coming from SVG when outside pathgroups
+   * @type Number
+   * @default 0
+   */
+  offsetX
```

Comment:
isnt' there a way with class definition to declare the default value in the code, with the property?